src/main/java/org/codehaus/mojo/exec/ExecMojo.java

```diff
@@ -488,6 +494,14 @@ private Map<String, String> handleSystemEnvVariables() throws MojoExecutionExcep
             }
         }
 
+        Toolchain tc = getToolchain();
+        if (tc != null) {
+            String toolchainJava = tc.findTool("java");
+            if (toolchainJava != null) {
+                enviro.put(toolchainJavaHomeEnvName, toolchainJava.replaceFirst("bin/java$", ""));
+            }
```

Comment:
here I also would check if `toolchain` is `jdk` - for other toolchains we don't know  if `bin` directory is used ....

Reply:
`replaceFirst("bin/java$", ""))`  on windows system we will have `java.exe`